packages/engine/src/simulation/package/state/packages/behavior_execution/package.js

```diff
@@ -171,14 +171,14 @@ const run_task = (experiment, sim, task_message, group_state, group_context) =>
 
     const n_agents_in_group = group_state.n_agents();
     for (var i_agent = 0; i_agent < n_agents_in_group; ++i_agent) {
-        
-        // Reuse `agent_state` and `agent_ctx` objects.
-        agent_state = group_state.get_agent(i_agent, agent_state);
+        // TODO: Reuse `agent_state` objects. When using the old `agent_state`, the indices for behaviors are borked
+        agent_state = group_state.get_agent(i_agent, null);
+        // Reuse `agent_ctx` objects.
         agent_ctx = group_context.get_agent(i_agent, agent_ctx);
 
         const behavior_ids = agent_state[behavior_ids_field_key];
         const n_behaviors = behavior_ids.length;
-        for (var i_behavior = agent_state.behavior_index; i_behavior < n_behaviors; ++i_behavior) {
+        for (var i_behavior = 0; i_behavior < n_behaviors; ++i_behavior) {
```

Comment:
This shouldn't always be 0, because the first behaviors of the agent might be in a different language (e.g. Python). Does using `agent_state.behavior_index` make a difference? Behavior indices should be reset to zero before running the package.

Reply:
This doesn't make a difference, probably everything is Javascript rightnow. Reverted the change.